Automattic/wp-calypso
# BackupRetention: Add retention setting in Calypso Blue

We released in #73035 the backup retention setting in Jetpack Cloud. After multiple iterations and enhancements, we are ready to include it in Calypso Blue.

## Proposed Changes

* Add backup retention setting in Calypso Blue
* Similar to #73568, update link on X days of backups saved to point to settings page with a cog icon in Jetpack Cloud and Calypso Blue.
* Enable the `jetpack/backup-retention-settings` feature in all environments.

## Demo

https://user-images.githubusercontent.com/1488641/222613753-78f2f414-7e05-4d60-bc12-a8a37f7bd690.mp4

## Screenshots

| Description | Screenshot |
|---|---|
| Setting page | ![image](https://user-images.githubusercontent.com/1488641/221468912-cbff13b6-4270-4018-a3d6-371cd2c1315d.png) |
| Confirmation dialog | ![image](https://user-images.githubusercontent.com/1488641/222614479-83388331-3330-476a-a5e2-b74d590f4097.png) |
| Tooltip | ![image](https://user-images.githubusercontent.com/1488641/222614427-0e63df41-b33a-4009-8d7d-076f390a6bed.png) |
| Success notice | ![image](https://user-images.githubusercontent.com/1488641/222615790-b69b8211-e320-43a1-a105-178e2b3d70cd.png) |
| Error notice | ![image](https://user-images.githubusercontent.com/1488641/222614689-2d56e8cd-8ea0-4f41-916e-f6530f255898.png) |
| Cloud storage space (link to settings) | ![image](https://user-images.githubusercontent.com/1488641/222629909-6ae61d2c-f15f-493a-8abf-06a1e54b3b34.png) |

## Testing Instructions



### Test backup retention setting

Unfortunately, this one cannot be tested using Calypso live branch, due to some restrictions redirecting back from checkout page to calypso.live domain.

* Checkout `update/backup-retention-calypso-blue` branch locally.
* Start Calypso blue using `ENABLE_FEATURES=no-force-sympathy yarn start`. This is to prevent state to being randomly cleared out when redirected back from checkout. More details on what sympathy means [here](https://github.com/Automattic/wp-calypso/pull/14121).
* Navigate to http://calypso.localhost:3000/ and switch to a site with a Jetpack VaultPress Backup plan with storage limit.
* Navigate to Settings > Jetpack.
* You should see the `Days of backups saved` and `Remote server credentials` settings.
* Play around with the `Days of backups saved` setting: increasing retention, decreasing retention, update setting, purchase and update retention and ensure it is working properly and similar as it does on Jetpack Cloud. We didn't change any of the functionality of the component due to this change, so we could focus on doing smoke tests on the functionality, specially in:
  * Purchase and update flow: ensure that after purchasing a storage addon, it redirects to Calypso Blue retentions setting.

### Test backup page link pointing to setting page

The idea is to validate on Calypso Blue and Jetpack Cloud that when visiting backup page and clicking on `X days of saved backups` link it points to its respecting setting page.

| Environment | Expected setting page URL | Screenshot |
|---|---|---|
| Calypso Blue | https://wordpress.com/settings/jetpack/[site] | ![image](https://user-images.githubusercontent.com/1488641/222631280-cfdb4962-ed8e-4eab-8f00-5b5046cb32cf.png) |
| Jetpack Cloud | https://cloud.jetpack.com/settings/[site] | ![image](https://user-images.githubusercontent.com/1488641/222631398-64b5d26b-d88b-4f9f-a57c-4b50ba274ebe.png) |


## Pre-merge Checklist
